src/proposals/20250807/SparkEthereum_20250807.t.sol

```diff
@@ -0,0 +1,285 @@
+        RateTargetKinkIRMParams memory oldParams = RateTargetKinkIRMParams({
+            irm                      : USDC_USDT_IRM_OLD,
+            baseRate                 : 0,
+            variableRateSlope1Spread : 0.01e27,
+            variableRateSlope2       : 0.15e27,
+            optimalUsageRatio        : 0.95e27
+        });
+        RateTargetKinkIRMParams memory newParams = RateTargetKinkIRMParams({
+            irm                      : USDC_USDT_PYUSD_IRM,
+            baseRate                 : 0,
+            variableRateSlope1Spread : 0.015e27,
+            variableRateSlope2       : 0.15e27,
+            optimalUsageRatio        : 0.95e27
+        });
+        _testRateTargetKinkIRMUpdate("USDC", oldParams, newParams);
+    }
+
+    function test_ETHEREUM_sparkLend_usdtIrmUpdate() public onChain(ChainIdUtils.Ethereum()) {
+        RateTargetKinkIRMParams memory oldParams = RateTargetKinkIRMParams({
+            irm                      : USDC_USDT_IRM_OLD,
+            baseRate                 : 0,
+            variableRateSlope1Spread : 0.01e27,
+            variableRateSlope2       : 0.15e27,
+            optimalUsageRatio        : 0.95e27
+        });
+        RateTargetKinkIRMParams memory newParams = RateTargetKinkIRMParams({
+            irm                      : USDC_USDT_PYUSD_IRM,
+            baseRate                 : 0,
+            variableRateSlope1Spread : 0.015e27,
+            variableRateSlope2       : 0.15e27,
+            optimalUsageRatio        : 0.95e27
+        });
+        _testRateTargetKinkIRMUpdate("USDT", oldParams, newParams);
+    }
+
+    function test_ETHEREUM_sparkLend_usdsFlashloanEnabled() public onChain(ChainIdUtils.Ethereum()) {
+        SparkLendContext memory ctx = _getSparkLendContext();
+
+        ReserveConfig[] memory allConfigsBefore = createConfigurationSnapshot('', ctx.pool);
+
+        assertEq(_findReserveConfigBySymbol(allConfigsBefore, 'USDS').isFlashloanable, false);
+
+        executeAllPayloadsAndBridges();
+
+        ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot('', ctx.pool);
+
+        assertEq(_findReserveConfigBySymbol(allConfigsAfter, 'USDS').isFlashloanable, true);
+    }
+
+    function test_ETHEREUM_sll_onboardPyusd() public onChain(ChainIdUtils.Ethereum()) {
+        bytes32 pyusdDepositKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(Ethereum.ALM_CONTROLLER).LIMIT_AAVE_DEPOSIT(),
+            PYUSD_ATOKEN
+        );
+        bytes32 pyusdWithdrawKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(Ethereum.ALM_CONTROLLER).LIMIT_AAVE_WITHDRAW(),
+            PYUSD_ATOKEN
+        );
+
+        _assertRateLimit(pyusdDepositKey,  0, 0);
+        _assertRateLimit(pyusdWithdrawKey, 0, 0);
+
+        executeAllPayloadsAndBridges();
+
+        _assertRateLimit(pyusdDepositKey,  50_000_000e6,      25_000_000e6 / uint256(1 days));
+        _assertRateLimit(pyusdWithdrawKey, type(uint256).max, 0);
```

Comment:
Add a deposit and withdraw transaction to make sure that the pyUSD integration works as a whole.

Reply:
Use `_testAaveOnboarding`